src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -413,14 +446,14 @@ private async Task ContainerHealthcheck(IExecutionContext executionContext, Cont
                 serviceHealth = (await _dockerManager.DockerInspect(context: executionContext, dockerObject: container.ContainerId, options: healthCheck)).FirstOrDefault();
                 retryCount++;
             }
-            if (string.Equals(serviceHealth, "healthy", StringComparison.OrdinalIgnoreCase))
-            {
-                executionContext.Output($"{container.ContainerNetworkAlias} service is healthy.");
-            }
-            else
-            {
-                throw new InvalidOperationException($"Failed to initialize, {container.ContainerNetworkAlias} service is {serviceHealth}.");
-            }
+            return serviceHealth;
+        }
+
+        private async Task ContainerErrorLogs(IExecutionContext executionContext, ContainerInfo container)
```

Comment:
do we need a method for these 3 lines? also, the method doesn't seem to be called more than once.

Reply:
It is a matter of taste. I personally think it is always a good practice to extract the code into smaller methods - it is easier to read.

Reply:
When I read the method name, my first reaction is we are getting a log from a failed container. however, the reality is we do not only get the log, we also log a user-facing issue and mutate the `container` object which doesn't line up with the method name.

Reply:
Ok, I moved the code block back to the loop and removed the method.